src/lua/datetime.lua

```diff
@@ -271,38 +287,16 @@ local function check_dt(dt_v, direction, v, unit)
     end
 end
 
-local function check_ymd_range(y, M, d)
```

Comment:
@ligurio @sergepetrenko I could not resist and not simplify this code after we have introduced `dt_from_ymd_checked` recently. That's not required change and could be reverted (if we are not ok with less concise error messages now), but it looks significantly simpler now.

Reply:
I like the idea.